src/Controller/AuthenticationController.php

```diff
@@ -166,9 +170,19 @@ public function login(Request $request): Response {
           $destination_url->setRouteParameter('changePassword', 'yes');
         }
       }
+      else if (isset($redirect)) {
+        $response_class = $redirect_class;
+        $destination_url = $redirect;
+      }
     }
 
-    return new $response_class($destination_url->setAbsolute()->toString());
+    if ($request->headers->has('x-requested-with') && 'XMLHttpRequest' == $request->headers->get('x-requested-with')) {
+      $response_class = AjaxResponse::class;
+      return new $response_class(new RedirectCommand($destination_url->setAbsolute()->toString()));
+    }
+    else {
```

Comment:
Given the if block above uses `return` the `else` here is redundant. The `return` below doesn’t need an `else` around it.

Reply:
I have removed this.